apps/files_sharing/lib/Controller/ShareController.php

```diff
@@ -712,8 +716,14 @@ protected function singleFileDownloaded(Share\IShare $share, \OCP\Files\Node $no
 		} else {
 			if ($node instanceof \OCP\Files\File) {
 				$subject = Downloads::SUBJECT_PUBLIC_SHARED_FILE_DOWNLOADED;
+				if (isset($remoteAddressHash)) {
+					$parameters[] = $remoteAddressHash;
+				}
```

Comment:
```suggestion
				$parameters[] = $remoteAddressHash;
```

isset should not be needed? It can never not be there. It's always an md5 of something?

Reply:
right, psalm said the same thing :D

Reply:
fixed